tests/test_kubernetes_tools.py

```diff
@@ -767,18 +767,43 @@ def test_should_run_uwsgi_exporter_sidecar_defaults(self):
             default_should_run_uwsgi_exporter_sidecar=mock.Mock(return_value=False)
         )
 
-        assert (
-            self.deployment.should_run_uwsgi_exporter_sidecar(
-                system_paasta_config_enabled
+        with mock.patch(
+            "paasta_tools.kubernetes_tools.DEFAULT_USE_PROMETHEUS_UWSGI",
+            autospec=True,
+            new=False,
+        ):
+            assert (
+                self.deployment.should_run_uwsgi_exporter_sidecar(
+                    system_paasta_config_enabled
+                )
+                is True
             )
-            is True
-        )
-        assert (
-            self.deployment.should_run_uwsgi_exporter_sidecar(
-                system_paasta_config_disabled
+            assert (
+                self.deployment.should_run_uwsgi_exporter_sidecar(
+                    system_paasta_config_disabled
+                )
+                is False
+            )
+
+        # If the default for use_prometheus is True and config_dict doesn't specify use_prometheus, we should run
+        # uwsgi_exporter regardless of default_should_run_uwsgi_exporter_sidecar.
+        with mock.patch(
+            "paasta_tools.kubernetes_tools.DEFAULT_USE_PROMETHEUS_UWSGI",
+            autospec=True,
+            new=True,
```

Comment:
```suggestion
```

Reply:
? These lines are important to the test. It's what's setting DEFAULT_USE_PROMETHEUS_UWSGI to True and False.

Reply:
(oh but I did forget to set autospec=False after the pre-commit complained about me not having an autospec line)

Reply:
ah, I didn't read the error message better than the tests and suggested removing the wrong line :p